src/AbstractUri.php

```diff
@@ -956,7 +1055,11 @@ public function withPort($port): self
      */
     protected static function filterPort($port)
     {
-        if (null !== $port && !is_port($port)) {
+        if (null === $port) {
+            return $port;
+        }
+
+        if ($port < 1) {
```

Comment:
I don't think there's anything that says a port must be > 0.

Reply:
no there's not ... but 

>  port        = *DIGIT

so maybe make `if ($port < 0)`

Reply:
Yes, that's better. `0` is e.g. used by Linux in `tcp://127.0.0.1:0` to bind to a random available port.